### PR TITLE
Migrate title section to heading

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Needs https://github.com/home-assistant/frontend/pull/22008

I extracted the logic into multiple function so we can add more migration in the future.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Release Notes

- **New Features**
  - Introduced validation for Lovelace configurations to ensure correctness and prevent issues from invalid settings.
  - New default card initialization for newly created sections in the Lovelace interface.

- **Improvements**
  - Removed the title property from section settings, streamlining configuration management.
  - Simplified user interface for section management by updating icons and reducing visual clutter.
  - Enhanced clarity in deletion confirmation messages for sections.

- **Deprecations**
  - Marked the `title` property in Lovelace section configuration as deprecated, recommending the use of heading cards instead.

### CODE REVIEW

File: src/panels/lovelace/views/hui-sections-view.ts [hunk 1]
```diff
@@ -256,7 +246,12 @@ export class SectionsView extends LitElement implements LovelaceViewElement {
   private _createSection(): void {
     const newConfig = addSection(this.lovelace!.config, this.index!, {
       type: "grid",
-      cards: [],
+      cards: [
+        {
+          type: "header",
+          title: "New Section",
```
Comment (piitaya) [inline src/panels/lovelace/views/hui-sections-view.ts:252]:
```suggestion
          type: "heading",
          heading: "New Section",
```

File: src/panels/lovelace/ha-panel-lovelace.ts [hunk 2]
```diff
@@ -321,26 +318,8 @@ export class LovelacePanel extends LitElement {
   }
 
   private _checkLovelaceConfig(config: LovelaceRawConfig) {
-    // Somehow there can be badges with value null, we remove those
-    if (isStrategyDashboard(config)) {
-      return config;
-    }
-    let checkedConfig = !Object.isFrozen(config) ? config : undefined;
```
Comment (piitaya) [inline src/panels/lovelace/ha-panel-lovelace.ts:328]: Not sure why the config can be frozen here. I have not been able to reproduce this case.

File: src/panels/lovelace/views/hui-sections-view.ts [hunk 3]
```diff
@@ -281,20 +276,18 @@ export class SectionsView extends LitElement implements LovelaceViewElement {
...
       const type = cardCount ? "cards" : "only";
 
       const confirm = await showConfirmationDialog(this, {
         title: this.hass.localize(
           "ui.panel.lovelace.editor.delete_section.title"
         ),
         text: this.hass.localize(
-          `ui.panel.lovelace.editor.delete_section.text_${named}_section_${type}`,
-          { name: title }
+          `ui.panel.lovelace.editor.delete_section.text_${named}_section_${type}`
```
Comment (bramkragten) [inline src/panels/lovelace/views/hui-sections-view.ts:290]: We can just change this now to this right?
```suggestion
          `ui.panel.lovelace.editor.delete_section.text_unnamed_section_${type}`
```

**Review by bramkragten — DISMISSED** [review]

File: src/panels/lovelace/ha-panel-lovelace.ts (hunk 2 above)
Comment (silamon) [reply src/panels/lovelace/ha-panel-lovelace.ts:328]: This is where it has been introduced: https://github.com/home-assistant/frontend/pull/4862

Comment (piitaya) [reply src/panels/lovelace/ha-panel-lovelace.ts:328]: Yeah followed by https://github.com/home-assistant/frontend/pull/4862. 
Now we always migrate and always freeze. It should not change a lot about performance, I checked with @bramkragten.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**